ssl-config/src/main/java/com/palantir/remoting/ssl/SslConfiguration.java

```diff
@@ -20,15 +20,29 @@
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import com.google.common.base.Optional;
 import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
 import java.nio.file.Path;
+import java.security.GeneralSecurityException;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManagerFactory;
 import org.immutables.value.Value;
 import org.immutables.value.Value.Style.ImplementationVisibility;
 
+/**
+ * Configuration that provides a {@link TrustManagerFactory} based on a trust store and optionally provides
+ * a {@link KeyManagerFactory} based on a key store. The stores can be in JKS or PKCS12 format.
+ */
 @JsonDeserialize(as = ImmutableSslConfiguration.class)
 @JsonSerialize(as = ImmutableSslConfiguration.class)
 @Value.Immutable
 @Value.Style(visibility = ImplementationVisibility.PACKAGE)
-public abstract class SslConfiguration {
+public abstract class SslConfiguration implements SslConfigurationProvider {
```

Comment:
I guess I was thinking that the file/directory based config would just be a puppet-style SSL config directory, and we'd set type to "puppet" and path to the base dir. 

Shouldn't actually need more config options?